server/src/diagnostic_ext.rs

```diff
@@ -28,24 +29,33 @@ pub async fn publish_diagnostics_for_set(
         .filter_map(|uri| Some((convert_slice_url_to_uri(uri)?, vec![])))
         .collect::<HashMap<Url, Vec<tower_lsp::lsp_types::Diagnostic>>>();
 
-    // Process the diagnostics and populate the map
-    process_diagnostics(diagnostics, &mut map);
+    // Process the diagnostics and populate the map. Any diagnostics that do not have a span are returned for further processing.
+    let spanless_diagnostics = process_diagnostics(diagnostics, &mut map);
+    for diagnostic in spanless_diagnostics.iter() {
```

Comment:
Also redundant and weird:
```suggestion
    for diagnostic in spanless_diagnostics {
```